pages/_document.js

```diff
@@ -32,7 +32,9 @@ export default class MyDocument extends Document {
   render() {
     return (
       <Html>
-        <Head />
+        <Head>
+          <link rel="shortcut icon" href="/images/favicon.ico" />
+        </Head>
```

Comment:
for some reason, the favicon only worked placed in a subfolder in the `public` folder. Other people have this issue too https://github.com/vercel/next.js/issues/4414